java/arcs/core/host/Utils.kt

```diff
@@ -36,9 +37,11 @@ fun KClass<*>.className(): String {
 
 /** Returns a pair mapping [ParticleIdentifier] to [ParticleConstructor] */
 inline fun <reified T : Particle> (() -> T).toRegistration(): ParticleRegistration =
-    T::class.toParticleIdentifier() to object : ParticleConstructor.Empty() {
-        override fun invoke(): Particle = this@toRegistration.invoke()
-    }
+    T::class.toParticleIdentifier() to { _ -> this.invoke() }
```

Comment:
Much better.

Reply:
Although I don't think we can get rid of requiring this to be a suspend function, because on DevHost, we will need to dynamically load code which may not be a synchronous operation.  That is, for ParticleRegistration on DevEx, this is probably going to send an async message to the Web runtime, tell it to load and instantiate the particle, and then return back some token.